compiler/data/lambda-class-data.cpp

```diff
@@ -40,6 +40,10 @@ LambdaPtr LambdaClassData::get_from(VertexPtr v) {
 }
 
 void LambdaClassData::infer_uses_assumptions(FunctionPtr parent_function) {
+  auto expected = AssumptionStatus::unknown;
+  if (!assumption_vars_status.compare_exchange_strong(expected, AssumptionStatus::processing)) {
+    return;
```

Comment:
Is it make sense to perform `assumption_busy_wait()` here?

Reply:
honestly no, we just need this condition to prevent inferring in the preprocess-functions pass if it's been already inferred before, due-to typed-lambdas.
all in all, everything should be fine